actions/setup-flutter-and-dart/action.yaml

```diff
@@ -0,0 +1,48 @@
+name: Setup Flutter and Dart
+description: |
+  Installs flutter & dart, using a non-flutter dart binary of specified channel
+
+inputs:
+  flutter-channel:
+    required: false
+    description: Flutter channel to install
+    default: "stable"
+  dart-channel:
+    required: false
+    description: Dart channel to install
+    default: stable
+
+runs:
+  using: composite
+  steps:
+    - name: Get latest Flutter stable commit
+      shell: bash
+      id: flutter-stable
+      run: |
+        sha=$(
+          curl -fsSL \
+            --retry 3 \
+            -H "Accept: application/vnd.github+json" \
+            -H "X-GitHub-Api-Version: 2022-11-28" \
+            https://api.github.com/repos/flutter/flutter/branches/${{ inputs.flutter-channel }} |
+          jq -r '.commit.sha'
+        )
+        echo "sha=$sha" >> "$GITHUB_OUTPUT"
```

Comment:
Since flutter takes long to setup, cache it against the current git ref for it's channel.

Reply:
Keep an eye on this, will potentially affect scorecard.

Reply:
Workflow to create PRs updating SHAs periodically